Add ".writing" suffix to raw data filenames during the time that they are being actively written

bieryAtFnal opened the pull request:
As part of this, I added a small "file handle" class to wrap the the HighFive::File instance and automatically rename the raw data files from `xyz.writing` to `xyz` when the file handle is destructed.  This necessitated a few changes in other places...

Thinking about this file handle class a little bit more now (as I write this), it might make sense to move the call to new HighFive::File from the `open_file_if_needed()` method to the HDF5FileHandle constructor, but I'll leave that for another time.

eflumerf commented on plugins/HDF5DataStore.hpp:
```diff
@@ -482,25 +524,47 @@ class HDF5DataStore : public DataStore
       m_basic_name_of_open_file = file_name;
       m_open_flags_of_open_file = open_flags;
-      m_file_ptr.reset(new HighFive::File(unique_filename, open_flags));
+      unique_filename += HDF5FileHandle::s_inprogress_filename_suffix;
```
Would it be better to have HDF5FileHandle::HDF5FileHandle take a string argument, so this becomes something like:
```CPP
try {
  m_file_handle.reset(new HDF5FileHandle(unique_filename, open_flags));
} catch {
```
and then you could store the original file name in HDF5FileHandle which makes the logic in its destructor much simpler...

bieryAtFnal commented:
Eric, I've made two changes.  I modified the HDF5FileHandle constructor as you suggested, and I changed the base branch for this PR from _develop_ to _patch/2.8.2_.  If you have a chance, I'd appreciate it if you can take a look.
Thanks

eflumerf reviewed (approved):
Looks good to me. Code builds and has correct behavior when running.
